src/package-request.js

```diff
@@ -154,36 +155,7 @@ export default class PackageRequest {
     name: string,
     range: string,
   } {
-    let hasVersion = false;
-    let range = 'latest';
-    let name = pattern;
-
-    // if we're a scope then remove the @ and add it back later
-    let isScoped = false;
-    if (name[0] === '@') {
-      isScoped = true;
-      name = name.slice(1);
-    }
-
-    // take first part as the name
-    const parts = name.split('@');
-    if (parts.length > 1) {
-      name = parts.shift();
-      range = parts.join('@');
-
-      if (range) {
-        hasVersion = true;
-      } else {
-        range = '*';
-      }
-    }
-
-    // add back @ scope suffix
-    if (isScoped) {
-      name = `@${name}`;
-    }
-
-    return {name, range, hasVersion};
+    return normalizePattern(pattern);
```

Comment:
It originally lived inside package-request, but bundling package-request blew up the bundle by 10x, so I moved it outside of package-request, but didn't want to upset the code that expected it to live on the class static method.

That was before I boldly found & replaced all the `wrapper.js`s in all the files. I could probably do the same here with `normalizePattern`